test/plugin/test_parser_json.rb

```diff
@@ -111,4 +111,29 @@ def test_parse_with_keep_time_key_without_time_format(data)
       assert_equal text, record['time']
     end
   end
+
+  data('yajl' => 'yajl')
+  def test_yajl_parse_io_with_buffer_smaller_than_input(data)
```

Comment:
I was considering adding a test case for the inverse of this (input smaller than buffer) but didn't do this because it seemed like it would be quite awkward to implement - due to needing to assert on the parser *not* emitting an event - and probably not provide much value.
Happy to reconsider this if anyone has a suggestion on how this could be done neatly though!